api/v1/announcements/PKPAnnouncementHandler.php

```diff
@@ -220,33 +219,42 @@ public function add($slimRequest, $response, $args)
             [$contextId]
         );
 
+        if ($sendEmail) {
+            $userIdsToMail = $notificationSubscriptionSettingsDao->getSubscribedUserIds(
+                [NotificationSubscriptionSettingsDAO::BLOCKED_NOTIFICATION_KEY, NotificationSubscriptionSettingsDAO::BLOCKED_EMAIL_NOTIFICATION_KEY],
+                [PKPNotification::NOTIFICATION_TYPE_NEW_ANNOUNCEMENT],
+                [$contextId]
+            );
+
+            list ($userIdsToNotify, $userIdsToNotifyAndMail) = [
+                $userIdsToNotify->diff($userIdsToMail),
+                $userIdsToNotify->intersect($userIdsToMail)
+            ];
```

Comment:
I've considered that option, splitting it differently, but the end solution was really hard to read. That's because a notification should be created first in order to add an unsubscribe footer to email.

Reply:
If you want to  keep the `list()` syntax, please add a comment to the code which explains what the resulting values of `$userIdsToNotify` and `$userIdsToNotifyAndMail` consists of.

Reply:
No, I meant the separation notifications from emails to be processed by the job. It's not a problem to remove the list syntax but to process emails independently from notifications